test/unit/stats/views/usage_test.rb

```diff
@@ -40,6 +40,42 @@ class DummyClass < Stats::Base
      end
   end
 
+  test '#usage raises InvalidParameterError when range is more than one year for day granularity' do
+    assert_raise Stats::InvalidParameterError do
+      @dummy.usage(@options.merge!(until: '2011-01-01'))
+    end
+  end
+
+
+  test '#usage raises InvalidParameterError when range is more than 90 days for hour granularity' do
+    assert_raise Stats::InvalidParameterError do
+      @dummy.usage(@options.merge!(until: '2010-04-01', granularity: :hour))
+    end
+  end
+
+  test '#usage raises InvalidParameterError when range is more than 10 years for month granularity' do
+    assert_raise Stats::InvalidParameterError do
+      @dummy.usage(@options.merge!(until: '2020-01-01', granularity: :month))
+    end
+  end
+
+  test "#usage won't raise an Error when range is less than one year for day granularity" do
+    assert_nothing_raised do
```

Comment:
`assert_nothing_raised` is noop Just remove it :)

And merge this test into the one which is failing. Like demonstrating:

```
test 'something' do
  1. with this params it works
  2. with this other params it fails
end
```